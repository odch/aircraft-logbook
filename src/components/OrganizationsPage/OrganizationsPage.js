import React from 'react'
import Header from '../../containers/HeaderContainer'
import OrganizationsList from '../../containers/OrganizationsListContainer'

const OrganizationsPage = () => (
  <React.Fragment>
    <Header />
    <OrganizationsList />
  </React.Fragment>
)

export default OrganizationsPage
