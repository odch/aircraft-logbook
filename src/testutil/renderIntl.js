import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
