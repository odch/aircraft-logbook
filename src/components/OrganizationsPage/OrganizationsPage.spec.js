import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderIntl from '../../testutil/renderIntl'
import OrganizationsPage from './OrganizationsPage'

describe('components', () => {
  describe('OrganizationsPage', () => {
    it('renders correctly', () => {
      const state = {
        firebase: {
          auth: {
            isEmpty: false
          }
        },
        firestore: {
          ordered: {
            organizations: [{ id: 'org1' }, { id: 'org2' }, { id: 'org3' }]
          }
        },
        app: {
          organizations: {
            createDialogOpen: false,
            createDialogData: {
              name: ''
            }
          }
        }
      }

      const store = configureStore()(state)

      // workaround for containers which use firestoreConnect()
      Object.assign(store, state)

      const tree = renderIntl(
        <Provider store={store}>
          <OrganizationsPage />
        </Provider>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
