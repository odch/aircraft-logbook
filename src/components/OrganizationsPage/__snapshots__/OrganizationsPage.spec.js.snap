// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components OrganizationsPage renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-11"
  >
    <div
      className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
    >
      <button
        aria-label="Menu"
        className="MuiButtonBase-root-50 MuiIconButton-root-44 MuiIconButton-colorInherit-45 Header-menuButton-3"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-49"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-53"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-62"
        />
      </button>
      <h2
        className="MuiTypography-root-69 MuiTypography-title-75 MuiTypography-colorInherit-89 Header-grow-2"
      >
        Open Digital Flugreisebuch
      </h2>
      <div>
        <button
          aria-haspopup="true"
          aria-owns={null}
          className="MuiButtonBase-root-50 MuiIconButton-root-44 MuiIconButton-colorInherit-45"
          data-cy="user-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-49"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-53"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-62"
          />
        </button>
      </div>
    </div>
  </header>,
  <div
    className="MuiGrid-container-105 MuiGrid-spacing-xs-24-129 OrganizationsList-container-99"
    data-cy="organizations-list"
  >
    <div
      className="MuiGrid-item-106 MuiGrid-grid-xs-12-145 MuiGrid-grid-sm-4-151"
    >
      <div
        className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-202 OrganizationsList-card-100"
      >
        <button
          className="MuiButtonBase-root-50 MuiCardActionArea-root-203 OrganizationsList-actionArea-102"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="MuiCardContent-root-206 OrganizationsList-cardContent-101"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-53 OrganizationsList-createIcon-103"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z"
              />
            </svg>
            <span
              className="MuiTypography-root-69 MuiTypography-button-80 OrganizationsList-createText-104"
            >
              <span>
                Neue Organisation erstellen
              </span>
            </span>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight-205"
          />
          <span
            className="MuiTouchRipple-root-62"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-item-106 MuiGrid-grid-xs-12-145 MuiGrid-grid-sm-4-151"
    >
      <div
        className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-202 OrganizationsList-card-100"
      >
        <button
          className="MuiButtonBase-root-50 MuiCardActionArea-root-203 OrganizationsList-actionArea-102"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="MuiCardContent-root-206 OrganizationsList-cardContent-101"
          >
            <h2
              className="MuiTypography-root-69 MuiTypography-title-75"
            >
              org1
            </h2>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight-205"
          />
          <span
            className="MuiTouchRipple-root-62"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-item-106 MuiGrid-grid-xs-12-145 MuiGrid-grid-sm-4-151"
    >
      <div
        className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-202 OrganizationsList-card-100"
      >
        <button
          className="MuiButtonBase-root-50 MuiCardActionArea-root-203 OrganizationsList-actionArea-102"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="MuiCardContent-root-206 OrganizationsList-cardContent-101"
          >
            <h2
              className="MuiTypography-root-69 MuiTypography-title-75"
            >
              org2
            </h2>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight-205"
          />
          <span
            className="MuiTouchRipple-root-62"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-item-106 MuiGrid-grid-xs-12-145 MuiGrid-grid-sm-4-151"
    >
      <div
        className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-202 OrganizationsList-card-100"
      >
        <button
          className="MuiButtonBase-root-50 MuiCardActionArea-root-203 OrganizationsList-actionArea-102"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="MuiCardContent-root-206 OrganizationsList-cardContent-101"
          >
            <h2
              className="MuiTypography-root-69 MuiTypography-title-75"
            >
              org3
            </h2>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight-205"
          />
          <span
            className="MuiTouchRipple-root-62"
          />
        </button>
      </div>
    </div>
  </div>,
]
`;
